platforms/nuttx/src/bootloader/common/image_toc.c

```diff
@@ -52,7 +52,7 @@
 
 bool find_toc(const image_toc_entry_t **toc_entries, uint8_t *len)
 {
-	const uintptr_t toc_start_u32 = APP_LOAD_ADDRESS + BOOT_DELAY_ADDRESS + 8;
+	const uintptr_t toc_start_u32 = FLASH_START_ADDRESS + BOOT_DELAY_ADDRESS + 8;
 	const image_toc_start_t *toc_start = (const image_toc_start_t *)toc_start_u32;
 	const image_toc_entry_t *entry = (const image_toc_entry_t *)(toc_start_u32 + sizeof(image_toc_start_t));
 
```

Comment:
No really related to this PR, bug while on it, could you also add the mitigation to jira DP-2949? Namely this:

- Apply a check to ensure that the TOC boundaries lie within the boundaries specified by the first entry.

No need to do the other "additional hardenings" or such mentioned in the jira.

For more info, see: https://ssrc.atlassian.net/browse/DP-2949

Just that if going to change/upstream the change, this saves some extra burden in the future.